### PR TITLE
Update docs links to use new URLs

### PR DESCRIPTION
**Pull request by hflynn**
See https://trello.com/c/oxWdJSKf/610-clean-up-remaining-links-to-site-support-old-docs-urls

### CODE REVIEW

**Comment by hflynn** [comment]
Will push another commit here to fix CONTRIBUTING.md too

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Contributing to Bio-Formats
 
 Guidance for contributing to OME projects in general can be found at
-http://www.openmicroscopy.org/site/support/contributing/
+https://docs.openmicroscopy.org/contributing/
 
 Developer documentation specifically for Bio-Formats is at
-http://www.openmicroscopy.org/site/support/bio-formats/developers/
+https://docs.openmicroscopy.org/latest/bio-formats/developers/
```
Comment (sbesson) [inline CONTRIBUTING.md]: This diff highlights the fact that this `CONTRIBUTING.md` and the ones in the other cpp repos should probably be reviewed but I do not know we want to do that here.

Comment (hflynn) [reply CONTRIBUTING.md]: Probably better to do on another round unless you just want me to delete most of this. There aren't really any developer focused C++ docs yet, are there?

Comment (sbesson) [reply CONTRIBUTING.md]: https://trello.com/c/I5VkCdRX/510-review-contributingmd created. I would say this is very low priority until we start getting external contributions to this repository (at which point we'll probably to point at or extend the relevant documentation sections)

**Pull request merged**